proposals/public-suffix.md

```diff
@@ -0,0 +1,820 @@
+
+Unfortunately, while this offered a solution to the performance problem,
+it added additional complexity to the API. To resolve this issue, the API
+has now been changed to being synchronous, which has allowed the batching method
+to be removed, thereby making the API more ergonomic.
+
+A high level analysis of the implementations of the major browser engines
+(Firefox, Chromium, Webkit) indicates that the synchronous approach is feasible,
+since the required functionality is already available in each browser engine's
+content/render process.
+
+#### 11. PSL Version
+
+Versioning metadata was introduced into the PSL with [this commit](https://github.com/publicsuffix/list/issues/1808#issuecomment-2455793503).
+
+Method `getVersion()` of this API should return the value of the
+VERSION metadata field contained in the specific PSL dataset used by the browser.
+
+### New Permissions
+
+| Permission Added | Suggested Warning |
+| ---------------- | ----------------- |
+| publicSuffix     | N/A               |
+
+Method `publicSuffix.getVersion()` may theoretically contribute to making the browser
+fingerprintable. However, it is already possible for malicious sites to get the browser's
+version, therefore this API does not introduce significant additional fingerprintability.
+
+### Manifest File Changes
+
+There are no changes to the manifest.
+
+## Security and Privacy
+
+### Exposed Sensitive Data
+
+The only data exposed by this API is the [public suffix list](https://publicsuffix.org/list/).
+
+### Abuse Mitigations
+
+This does not expose any new non-public data, nor does it significantly increase
+browser fingerprintability, so there are no new abuse vectors.
+
+### Additional Security Considerations
+
+N/A
+
+## Alternatives
+
+### Existing Workarounds
+
+Developers can download the PSL dataset, bundle it with their extensions, and implement
+logic that parses and interprets the dataset in order to determine the
+registrable domain for a domain name. There are several drawbacks to
+this approach:
+
+1. Potential for inconsistencies in the determination of registrable domains
+by the extension and the host browser, due to differences in the version
+of the PSL dataset used, and differences in the implementations that the host
+browser and the extension use in order to interpret this dataset.
+2. Increased complexity and maintenance costs of extensions.
+3. Increased performance overhead of extensions due to bundling of bulky
+PSL dataset, leading to increase in memory, disk usage and increase in CPU usage
+due to possibly suboptimal extension implementations and repeating work already
+done by the host browser.
+
+### Open Web API
+
+The purpose of this API is to eliminate the potential for inconsistency between
+the host browser and its hosted extensions. The simplest way of achieving this
+is for extensions to access this functionality via the host browser itself rather
+than via some external source, such as an Open Web API.
```

Comment:
By "Open Web API", we mean any API implemented by the browser and exposed in JavaScript. These can absolutely use internal sources of data, so I don't think this is a compelling argument. I would perhaps update this to focus more on the fact that:

- There is a higher bar for introducing web APIs, and we haven't seen much interest in moving forward a proposal like this in the past. Therefore, we would prefer to start with extensions, and can always propose a web API later if this work proves useful and there is appetite.
- The PSL is not appropriate for use in all circumstances. Extensions have a very compelling set of use cases that match browser use cases, but there is not a universal agreement this is the case more generally.

Reply:
Addressed in f1d81e5c10e93748e49c7a018ac9351b404fdd6d - I have updated this section as you suggest

Reply:
Thanks!